var/spack/repos/builtin/packages/lcals/package.py

```diff
@@ -0,0 +1,134 @@
+       (See the 1986 technical report: "The Livermore Fortran Kernels:
+        A Computer Test of the Numerical Performance Range",
+        by Frank H. McMahon, UCRL-53745.). The suite contains facilities to
+        generate timing statistics and reports."""
+
+    homepage = "https://codesign.llnl.gov/LCALS-downloads/"
+    url = "https://codesign.llnl.gov/LCALS-downloads/lcals-v1.0.2.tgz"
+
+    tags = ['proxy-app']
+
+    version('1.0.2', '40c65a88f1df1436a2f72b7d3c986a21')
+
+    variant(
+        'microarch',
+        description='Micro arch: SSE, AVX, MIC.',
+        default='sse',
+        values=('sse', 'avx', 'MIC'),
+    )
+
+    @property
+    def build_targets(self):
+
+        targets = []
+
+        cxxflags = '-std=c++0x '
+        cxx_compile = ''
+
+        microarch = self.spec.variants['microarch'].value
+
+        arch = platform.machine()
+
+        if microarch == 'MIC':
+            arch = 'MIC'
+        elif arch == 'x86_64' or arch == 'x86_32':
+            arch = 'x86'
+        elif arch != 'bgq':
+            raise InstallError('unknown architecture.')
+
+        if self.compiler.name == 'intel':
+            if arch == 'MIC':
+                cxxflags += '-DLCALS_PLATFORM_X86_SSE -DLCALS_COMPILER_ICC '
+                cxx_compile += '-g -O3 -mmic -vec-report3 '
+                ' -inline-max-total-size=10000 -inline-forceinline -ansi-alias'
+            elif microarch == 'sse' and arch == 'x86':
+                cxxflags += '-DLCALS_PLATFORM_X86_SSE -DLCALS_COMPILER_ICC '
+                cxx_compile += '-O3 -msse4.1 -inline-max-total-size=10000'
+                ' -inline-forceinline -ansi-alias -std=c++0x '
+            elif microarch == 'avx' and arch == 'x86':
+                cxxflags += '-DLCALS_PLATFORM_X86_AVX -DLCALS_COMPILER_ICC '
+                cxx_compile += '-O3 -mavx -inline-max-total-size=10000'
+                ' -inline-forceinline -ansi-alias -std=c++0x'
+            cxxflags += self.compiler.openmp_flag
+        elif self.compiler.name == 'gcc':
+            if arch == 'MIC' or (microarch == 'sse' and arch == 'x86'):
+                cxxflags += '-DLCALS_PLATFORM_X86_SSE -DLCALS_COMPILER_GNU '
+                cxx_compile += '-Ofast -msse4.1 -finline-functions'
+                ' -finline-limit=10000 -std=c++11 '
+                print('Im here!!!!')
+            elif microarch == 'avx' and arch == 'x86':
+                cxxflags += '-DLCALS_PLATFORM_X86_AVX -DLCALS_COMPILER_GNU '
+                cxx_compile += '-Ofast -mavx -finline-functions'
+                ' -finline-limit=10000 -std=c++11'
+            elif arch == 'bgq':
+                cxxflags += '-DLCALS_PLATFORM_BGQ -DLCALS_COMPILER_GNU '
+                cxx_compile += '-O3 -finline-functions -finline-limit=10000'
+                ' -std=c++0x'
+            cxxflags += self.compiler.openmp_flag
+        elif self.compiler.name == 'xl' and arch == 'bgp':
+            if self.compiler.version == 9 and arch == 'bgp':
```

Comment:
Did you check that this actually works? You might have to do `Version('9')` instead, I'm not sure.